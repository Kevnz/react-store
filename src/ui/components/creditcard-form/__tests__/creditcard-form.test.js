import React from 'react'
import { render, fireEvent } from 'react-testing-library'
import CreditCardForm from '../creditcard-form'
describe('The Credit Card Form', () => {
  it('should render', () => {
    const mockCB = jest.fn()
    const { container } = render(<CreditCardForm onSubmit={mockCB} />)
    expect(container).toMatchSnapshot()
  })
  it('should handle a cancel', () => {
    const mockCB = jest.fn()
    const { container } = render(
      <CreditCardForm onSubmit={() => {}} onCancel={mockCB} />
    )
    const btn = container.querySelector('button.primary')
    const cancelBtn = container.querySelector('button.default')
    expect(btn).toMatchSnapshot()
    expect(cancelBtn).toMatchSnapshot()
    fireEvent.click(cancelBtn)
    expect(mockCB).toHaveBeenCalled()
  })
})
