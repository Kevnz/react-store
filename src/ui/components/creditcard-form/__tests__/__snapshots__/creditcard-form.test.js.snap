// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Credit Card Form should handle a cancel 1`] = `
<button
  class="primary"
>
  Purchase
</button>
`;

exports[`The Credit Card Form should handle a cancel 2`] = `
<button
  class="default"
>
  Cancel
</button>
`;

exports[`The Credit Card Form should render 1`] = `
<div>
  <div
    class="checkout"
  >
    <form
      class="rfe-form"
      name="ccform"
    >
      <div
        class="checkout-header"
      >
        <h4
          class="checkout-title"
        >
          Checkout
          <span
            class="checkout-price"
          >
            $
          </span>
        </h4>
      </div>
      <div
        name="Row"
      >
        <input
          class="name"
          id="id-1-1"
          name="nameOnCard"
          placeholder="Your name"
          required=""
          type="text"
          value=""
        />
        <input
          class="exp"
          id="id-2-1"
          name="mm"
          placeholder="MM"
          required=""
          type="text"
          value=""
        />
        <input
          class="exp"
          id="id-3-1"
          name="yy"
          placeholder="YY"
          required=""
          type="text"
          value=""
        />
      </div>
      <br />
      <div
        name="Row"
      >
        <input
          class="card"
          id="id-4-1"
          name="card"
          placeholder="4111 1111 1111 1111"
          required=""
          type="text"
          value=""
        />
        <input
          class="cvc"
          id="id-5-1"
          name="cvc"
          placeholder="CVC"
          required=""
          type="text"
          value=""
        />
      </div>
      <p>
        <button
          class="primary"
        >
          Purchase
        </button>
        <button
          class="default"
        >
          Cancel
        </button>
      </p>
    </form>
  </div>
</div>
`;
